extensions/wikia/CuratedContent/CuratedContentValidator.class.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+class CuratedContentValidator {
+	const LABEL_MAX_LENGTH = 48;
+
+	private $errors;
+	private $titles;
+	private $hadEmptyLabel;
+
+	public function __construct( $data ) {
+		$this->errors = [ ];
+		$this->titles = [ ];
+		$this->hadEmptyLabel = false;
+
+		// validate sections
+		foreach( $data as $section ) {
+			if ( !empty( $section['featured'] ) ) {
+				$this->validateFeaturedSection( $section );
+			} else {
+				$this->validateSection( $section );
+			}
+		}
+		// also check section for duplicate title
+		foreach( array_count_values( $this->titles ) as $title => $count ) {
+			if ( $count > 1 ) {
+				$this->error( [ 'title' => $title ], 'duplicatedLabel' );
+			}
+		}
+	}
+
+	private function error( $itemWithTitle, $errorString ) {
+		if ( array_key_exists( 'title', $itemWithTitle ) && !empty( $errorString ) ) {
+			$this->errors[ ] = [ 'title' => $itemWithTitle['title'], 'reason' => $errorString ];
+		}
+	}
+
+	public function getErrors() {
+		return $this->errors;
+	}
+
+	private function validateFeaturedSection( $section ) {
+		foreach ( $section['items'] as $item ) {
+			$this->validateItem( $item );
+		}
+	}
+
+	private function validateImage( $sectionOrItem ) {
+		if ( empty( $sectionOrItem['image_id'] ) ) {
+			$this->error( $sectionOrItem, 'imageMissing' );
+		}
+	}
+
+	private function validateSection ( $section ) {
+		if ( empty( $section['title'] ) ) {
+			if ($this->hadEmptyLabel) {
```

Comment:
Also let's add a comment that this is "optional block" and that there can be only one.